src/pages/EditScholarship.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import firebase from 'firebase';
+import { Container, Typography, Alert, AlertTitle } from '@mui/material';
+import ScholarshipForm from '../components/ScholarshipForm';
+import SubmissionAlert from '../components/SubmissionAlert';
+import Scholarships from '../models/Scholarships';
+import { Helmet } from 'react-helmet';
+
+function EditScholarship() {
+  const { id } = useParams();
+  const [scholarship, setScholarship] = useState(undefined);
+  const [error, setError] = useState();
+  const loading = !error && (!scholarship || !scholarship.data);
+  const [submissionAlert, setSubmissionAlert] = useState(null);
+
+  // Fetch the scholarship if we need to load it
+  useEffect(() => {
+    let mounted = true;
+    if (loading) {
+      Scholarships.id(id)
+        .get()
+        .then((s) => mounted && setScholarship(s))
+        .catch((e) => mounted && setError(e));
+    }
+    return () => {
+      mounted = false;
+    };
+  }, [id, loading]);
```

Comment:
"if we need to load it" and the `if (loading)` checks come from `ViewScholarship` relying on passed in state. I think we can keep this simple and remove the effect's dependence on `loading`.

Reply:
nice, I thought we still needed it to show the '...Loading' screen but we can also show it when `error || !scholarship`